### ISSUE TITLE
The param sync about MMDistributedDataParallel

### ISSUE THREAD

**Issue by as754770178** [open-mmlab/mmcv#166]
I view the code in `MMDistributedDataParallel`, and have a question about param sync. The `forward` funtion exec param sync every time in pytorch `DistributedDataParallel`. The code as below:
```
    def forward(self, *inputs, **kwargs):
        if self.require_forward_param_sync:
            self._sync_params()
       ...  ...
```
But the `forward` funtion not exec param sync in pytorch `MMDistributedDataParallel`, the code as below:
```
    def forward(self, *inputs, **kwargs):
        inputs, kwargs = self.scatter(inputs, kwargs,
                                      [torch.cuda.current_device()])
        return self.module(*inputs[0], **kwargs[0])
```.
Can I think that if I use `MMDistributedDataParallel`, every node train independently, the param in  every node is different?

**Comment by hellock** [comment]
There is no synchronization during the forward process. Gradients will be synchronized after BP.

**Comment by as754770178** [comment]
I want to know the code about `Gradients will be synchronized after BP.` in where?
